Assets/_BForBoss/_Core/Scripts/Labor/RingSystem/RingBehaviour.cs

```diff
@@ -9,10 +9,22 @@ public class RingBehaviour : MonoBehaviour
         public Action<RingBehaviour> RingActivated;
         
         [SerializeField] private int _label;
+        [SerializeField] private Renderer _renderer;
 
         private TMPro.TMP_Text _labelTMP;
         private PlayerTriggerBehaviour _trigger;
 
+        public void SetLabel(string label)
+        {
+            _labelTMP.text = label;
+        }
+
+        public void SetColor(Color color)
+        {
+            if (_renderer == null) return;
```

Comment:
instead of doing null check every time, Do a Panic to check for the renderer on Awake